receiver/prometheusreceiver/internal/otlp_transaction.go

```diff
@@ -102,9 +105,44 @@ func (t *transaction) Append(ref storage.SeriesRef, labels labels.Labels, atMs i
 }
 
 func (t *transaction) AppendExemplar(ref storage.SeriesRef, l labels.Labels, e exemplar.Exemplar) (storage.SeriesRef, error) {
+	metricName := l.Get(model.MetricNameLabel)
+	familyName := normalizeMetricName(metricName)
+	if f, ok := t.metricBuilder.families[familyName]; ok {
+		gk := f.getGroupKey(l)
+		mg := f.groups[gk]
+		exemplar, exists := mg.exemplars[e.Value]
+		if !exists {
+			exemplar = pmetric.NewExemplar()
```

Comment:
if it already exists, we can just return.  We've already seen an exemplar at that value, so we would only keep one of them anyways.  We might as well not do the extra work, and just keep the first.